### PR TITLE
feat: make configgrpc support more encodings

### LINKED ISSUE #2548: configgrpc.go should support more encoding/compressions
**Is your feature request related to a problem? Please describe.**
Gzip is CPU intensive and offers less of a compression benefit than more modern compression like snappy or zstd

**Describe the solution you'd like**
https://github.com/mostynb/go-grpc-compression should be supported, not just gzip encoding, at https://github.com/open-telemetry/opentelemetry-collector/blob/68c5961f7bc2d8d55fd431ab07367fc16f8b26d0/config/configgrpc/configgrpc.go#L45

**Describe alternatives you've considered**
Do nothing, support gzip only

### PR DESCRIPTION
**Pull request by naseemkullah**
add support for lz4, snappy and zstd by means of
https://github.com/mostynb/go-grpc-compression to configgrcpc.go's
grpc client compression setting

fixes #2548

To test, I ran the docker-compose with a locally built image and configured the exporter with `lz4` compression. No error was emitted:

```yaml
...
exporters:
  otlp:
    endpoint: "otel-collector:4317"
    insecure: true
    compression: lz4
 ...
```

### CODE REVIEW

File: config/configgrpc/README.md [hunk 1]
```diff
@@ -16,7 +16,8 @@ configuration. For more information, see [configtls
 README](../configtls/README.md).
 
 - [`balancer_name`](https://github.com/grpc/grpc-go/blob/master/examples/features/load_balancing/README.md)
-- `compression` (default = gzip): Compression type to use (only gzip is supported today)
+- `compression`: compression to use among `gzip` (default), `lz4`,
```
Comment (bogdandrutu) [inline config/configgrpc/README.md:19]: Not sure where "default" is set.

File: exporter/otlpexporter/cfg-schema.yaml [hunk 2]
```diff
@@ -62,9 +62,10 @@ fields:
     https://github.com/grpc/grpc/blob/master/doc/naming.md.
 - name: compression
   kind: string
+  default: gzip
```
Comment (bogdandrutu) [inline exporter/otlpexporter/cfg-schema.yaml:65]: I don't think we set by default "gzip"

**Comment by naseemkullah** [comment]
bump

**Comment by tigrannajaryan** [comment]
@bogdandrutu @naseemkullah is this ready to be merged?

**Comment by naseemkullah** [comment]
yes @tigrannajaryan as far as I'm concerned

**Comment by alolita** [comment]
@naseemkullah pl fix these conflicts and resubmit.

**Comment by punya** [comment]
@bogdandrutu can we close this out (or allow the stale bot to perform it's life mission)? From today's SIG meeting, we couldn't find anybody to take over ownership. People can always reopen when they are motivated to work on it.

**Pull request closed without merging**